certbot-apache/certbot_apache/configurator.py

```diff
@@ -94,26 +95,49 @@ class ApacheConfigurator(augeas_configurator.AugeasConfigurator):
 
     OS_DEFAULTS = dict(
         server_root="/etc/apache2",
-        vhost_root="/etc/apache2/sites-available",
+        vhost_root="/etc/apache2/sites-enabled",
         vhost_files="*",
         logs_root="/var/log/apache2",
+        ctlpath="apache2ctl",
+        binpath="apache2",
         version_cmd=['apache2ctl', '-v'],
         apache_cmd="apache2ctl",
         restart_cmd=['apache2ctl', 'graceful'],
         conftest_cmd=['apache2ctl', 'configtest'],
         enmod=None,
         dismod=None,
         le_vhost_ext="-le-ssl.conf",
-        handle_mods=False,
+        handle_modules=False,
         handle_sites=False,
         challenge_location="/etc/apache2",
         MOD_SSL_CONF_SRC=pkg_resources.resource_filename(
             "certbot_apache", "options-ssl-apache.conf")
     )
 
-    def constant(self, key):
-        """Get constant for OS_DEFAULTS"""
-        return self.OS_DEFAULTS.get(key)
+    def option(self, key):
+        """Get a value from options"""
+        return self.options.get(key)
+
+    def _prepare_options(self):
+        """
+        Set the values possibly changed by command line parameters to
+        OS_DEFAULTS constant dictionary
+        """
+        opts = ["enmod", "dismod", "le_vhost_ext", "server_root", "vhost_root",
+                "logs_root", "challenge_location", "handle_modules", "handle_sites",
+                "ctlpath", "binpath"]
```

Comment:
I don't think it's easily achievable, as `add_parser_arguments` is a classmethod, and the changes made to it do not make it to the actual instance we'd be using when doing `prepare()`. One way to avoid this would be to add a `list` of `dict`s, each of which describe an command line argument as a class constant, but I feel that this would obscure the behavior more. For now, I added a comment reminding of changing the list in `_prepare_options()` as well. Do you feel that this would be enough?

Reply:
Why doesn't my proposal work? I suggested we make it a `set` to avoid issues if `add_parser_arguments` is called more than once and suggested making the `set` a global or class variable to work around `add_parser_arguments` being a class method. Am I missing something?

I'm still interested in an approach like this as it seems relatively simple and makes it easy to do things right in the future, but adding a comment is fine for now if you prefer.

Reply:
It's because `add_parser_arguments` are called on the class itself, and not an instance of it, so it doesn't allow us to store the added values to itself. Global variable could work, but I'm kind of thinking global variables largely as antipattern, my thinking might be flawed though.

Reply:
I was specifically referring to a class variable like `OS_DEFAULTS` instead an instance variable to work around this.

It's all a tradeoff though. Variables with a large scope can cause problems but so can trying to remember to properly make changes to multiple places in the future. A comment is fine for now.